src/docfx/lib/log/Report.cs

```diff
@@ -33,28 +33,27 @@ internal sealed class Report : IDisposable
 
         public int SuggestionCount => _suggestionCount;
 
-        public Report(bool legacy = false)
+        public Report(string docset = ".", bool legacy = false)
         {
+            _docsetPath = docset;
             _legacy = legacy;
+            _config = new Config();
+            _output = new Lazy<TextWriter>(() =>
+            {
+                // add default build log file output path
+                var outputFilePath = Path.GetFullPath(Path.Combine(_docsetPath, _config.Output.Path, "build.log"));
+
+                PathUtility.CreateDirectoryFromFilePath(outputFilePath);
+
+                return File.CreateText(outputFilePath);
+            });
         }
 
-        public void Configure(string docsetPath, Config config)
+        public void Configure(Config config)
         {
-            var outputPath = Path.Combine(docsetPath, config.Output.Path, "build.log");
-            Debug.Assert(_outputPath is null || _outputPath == outputPath, "Cannot change report output path");
+            Debug.Assert(!_output.IsValueCreated || _config.Output.Path == config.Output.Path, "Cannot change report output path");
 
             _config = config;
```

Comment:
With this change, If I set `output.path` in config, that config will not be respected